File: CategoryA/1845A.py
t = int(input())

for _ in range(t):
	n, k, x = map(int, input().split())
	
	if x != 1:
		print("YES")
		print(n)
		l = [1]*n
		print(*l)
	else:
		if k ==1:
			print("NO")
		elif k==2:
			if n%2 == 0:
				print("YES")
				print(n//2)
				p = [2] * (n//2)
				print(*p)
			else:
				print("NO")
		else:
			if n%2 == 0:
				print("YES")
				print(n//2)
				q = [2] * (n//2)
				print(*q)
			else:
				print("YES")
				print(n//2)
				print("3 " + "2 " * ((n - 3) // 2))	

File: CategoryA/1845A.c
#include <stdio.h>

int main() {
    int t;
    scanf("%d", &t);

    while (t--) {
        int n, k, x;
        scanf("%d %d %d", &n, &k, &x);

        if (x != 1) {
            printf("YES\n");
            printf("%d\n", n);
            for (int i = 0; i < n; i++) {
                printf("1");
                if (i < n - 1) printf(" ");
            }
            printf("\n");
        } else {
            if (k == 1) {
                printf("NO\n");
            } else if (k == 2) {
                if (n % 2 == 0) {
                    printf("YES\n");
                    printf("%d\n", n / 2);
                    for (int i = 0; i < n / 2; i++) {
                        printf("2");
                        if (i < n / 2 - 1) printf(" ");
                    }
                    printf("\n");
                } else {
                    printf("NO\n");
                }
            } else { // k >= 3
                if (n % 2 == 0) {
                    printf("YES\n");
                    printf("%d\n", n / 2);
                    for (int i = 0; i < n / 2; i++) {
                        printf("2");
                        if (i < n / 2 - 1) printf(" ");
                    }
                    printf("\n");
                } else {
                    printf("YES\n");
                    printf("%d\n", n / 2);
                    printf("3");
                    if ((n - 3) / 2 > 0) printf(" ");
                    for (int i = 0; i < (n - 3) / 2; i++) {
                        printf("2");
                        if (i < (n - 3) / 2 - 1) printf(" ");
                    }
                    printf("\n");
                }
            }
        }
    }

    return 0;
}

